src/services/content_processor.py

```
@@ -0,0 +1,306 @@
+                break
+            backtick_positions.append(pos)
+            pos += 3
+
+        # Process pairs of backticks
+        i = 0
+        while i < len(backtick_positions) - 1:
+            start_pos = backtick_positions[i]
+            end_pos = backtick_positions[i + 1]
+
+            # Extract the content between backticks
+            code_section = markdown_content[start_pos + 3 : end_pos]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)